applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1753 @@
+         * @see GenerateFaces()
+         */
+        SizeType EdgesNumber() const override
+        {
+            return 3;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * @brief This method gives you all edges of this geometry.
+         * @details This method will gives you all the edges with one dimension less than this geometry.
+         * For example a triangle would return three lines as its edges or a tetrahedral would return four triangle as its edges but won't return its six edge lines by this method.
+         * @return GeometriesArrayType containes this geometry edges.
+         * @see EdgesNumber()
+         * @see Edge()
+         */
+        GeometriesArrayType GenerateEdges() const override
+        {
+            GeometriesArrayType edges = GeometriesArrayType();
+            edges.push_back(Kratos::make_shared<EdgeType>(this->pGetPoint(0), this->pGetPoint(3), this->pGetPoint(4), this->pGetPoint(5), this->pGetPoint(1)));
+            edges.push_back(Kratos::make_shared<EdgeType>(this->pGetPoint(1), this->pGetPoint(6), this->pGetPoint(7), this->pGetPoint(8), this->pGetPoint(2)));
+            edges.push_back(Kratos::make_shared<EdgeType>(this->pGetPoint(2), this->pGetPoint(9), this->pGetPoint(10), this->pGetPoint(11), this->pGetPoint(0)));
+            return edges;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        SizeType FacesNumber() const override
+        {
+            return 3;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        //Connectivities of faces required
+        void NumberNodesInFaces(DenseVector<unsigned int>& NumberNodesInFaces) const override
+        {
+            if (NumberNodesInFaces.size() != 3) NumberNodesInFaces.resize(3, false);
+            NumberNodesInFaces[0] = 5;
+            NumberNodesInFaces[1] = 5;
+            NumberNodesInFaces[2] = 5;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        void NodesInFaces(DenseMatrix<unsigned int>& NodesInFaces) const override
+        {
+            // faces in columns
+            if (NodesInFaces.size1() != 6 || NodesInFaces.size2() != 3)
+                NodesInFaces.resize(6, 3, false);
+
+            //compatible to Edges() function. Triangle library considers a different order
+            NodesInFaces(0, 0) = 0;//face or master node
```

Comment:
it is not compatible with the ordering mentioned in this issue: [https://github.com/KratosMultiphysics/Kratos/issues/9990](url)

Reply:
Still not clear to me. I see the order is different than the order in triangular element, as it is done for 2D6N. I will investigate it in mode details.

Reply:
I just meant that the ordering that you mentioned in the above link is different from what is implemented here. Maybe you used another source. This was just a minor remark.